tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsWriter.cs

```diff
@@ -172,4 +205,25 @@ private async Task ProcessQueueLoopAsync()
             _processingMutex.Reset();
         }
     }
+
+    private void HandleConfigUpdate(AgentConfiguration config)
+    {
+        var isSupported = string.IsNullOrEmpty(config.DataStreamsMonitoringEndpoint)
+                              ? SupportState.Unsupported
+                              : SupportState.Supported;
+        var wasSupported = (SupportState)Volatile.Read(ref _isSupported);
+
+        if (isSupported != wasSupported)
+        {
+            _isSupported = (int)isSupported;
+            if (isSupported == SupportState.Supported)
+            {
+                Log.Information("Data streams monitoring supported, enabling flush");
+            }
+            else
+            {
+                Log.Warning("Data streams monitoring was enabled but is not supported by the Agent. Disabling flush");
```

Comment:
We could maybe even add: "Consider upgrading the agent to v{XXX} or disable DSM setting {envVarName} to false.